### PR TITLE
feat: propagate AgentTeams context in qoder and codex traces

### PR DESCRIPTION
**Pull request by Sunrisea**
## Summary

- Propagate `AGENTTEAMS_WORKER_NAME` and `AGENTTEAMS_INSTANCE_ID` into Qoder trace records as OTLP `resourceAttributes`
- Persist the same AgentTeams resource context through Codex Stop wakeup markers and project it onto transcript-derived trace records
- Align Qoder/Codex behavior with Claude Code by also setting `gen_ai.agent.name` from `agentteams.worker.name`

## Test

- `npm test -- tests/integration/hook-jsonl-flow.test.ts tests/unit/hooks/codex/hook-processor.test.mjs tests/unit/inputs/codex-transcript/codex-transcript-input.test.ts`
- `npm run typecheck`

### CODE REVIEW

**Review by ralf0131 — APPROVED** [review]
LGTM. Clean, well-structured change with thorough test coverage.

### Summary
Propagates `AGENTTEAMS_WORKER_NAME` / `AGENTTEAMS_INSTANCE_ID` into Qoder and Codex trace records as OTLP `resourceAttributes`, and aligns both with Claude Code by setting `gen_ai.agent.name` from `agentteams.worker.name`.

### Review Notes
- **Security:** `AGENTTEAMS_TOKEN` is explicitly tested to not leak; only whitelisted keys (`agentteams.worker.name`, `agentteams.instance.id`) are extracted from wakeup markers; `safeWakeupSessionPart` guards against path traversal in marker filenames; `collectResourceAttributesFromEnv` filters sensitive field names and enforces a 512-char max.
- **Backward compatibility:** `RESOURCE_ATTRIBUTE_FIELDS` is `{}` when no AgentTeams env vars are set, preserving existing behavior. The `finalizeRecords` refactor keeps the previous `cwd` application intact.
- **Tests:** Unit tests for both hook processors + integration test covering the end-to-end flow, including edge cases (empty values, non-string values, token filtering). CI green, typecheck passing.
- No issues found across correctness, performance, tests, or compatibility dimensions.

---
*Automated review by github-manager-bot*

File: src/inputs/codex-transcript/codex-transcript-input.ts [hunk 1]
```diff
@@ -22,6 +22,10 @@ import { stringValue, timestampMs } from './codex-transcript-utils.js';
 
 const DEFAULT_SESSION_DIR = '~/.codex/sessions';
 const READ_CHUNK_SIZE = 1024 * 1024;
+const WAKEUP_RESOURCE_ATTRIBUTE_KEYS = [
```
Comment (linrunqi08) [inline src/inputs/codex-transcript/codex-transcript-input.ts:27]: **[Medium]** `WAKEUP_RESOURCE_ATTRIBUTE_KEYS` 独立硬编码了两个 key,与共享模块 `resource-context.mjs` 的 `DEFAULT_RESOURCE_ENV_FIELD_MAP` 是两份互不感知的「真理来源」。

**影响:** 未来给共享 env map 增字段(如 region)时,Qoder 会透传该字段,而 Codex 会在此处静默丢弃 → 两端 trace 归属维度分叉,且这种缺失很难被察觉(隐性遥测缺口)。当前两侧行为一致,仅为潜在维护陷阱。

**建议:** 在此处加注释显式关联 `DEFAULT_RESOURCE_ENV_FIELD_MAP`(说明新增字段需同步两处),或后续将白名单从共享定义派生,避免双份维护。

**Comment by linrunqi08** [comment]
## 🔍 Code Review Summary

| Severity | Count |
|----------|-------|
| Critical | 0 |
| High     | 0 |
| Medium   | 1 |
| Low      | 2 |

### Lifecycle Verdict

| Check | Result |
|-------|--------|
| 资源释放 | PASS |
| 死锁/卡死风险 | PASS |
| 状态恢复正确性 | PASS |

### 总体结论

**未发现阻断合入的问题。** 实现干净、测试充分、安全意识良好：三层敏感字段过滤（env 名过滤 + 白名单 + 下游 key 过滤）、fail-open 读取、Qoder 所有记录构建路径统一收敛到 `finalizeRecords()`。本地已复现 `npm run typecheck` 通过、`codex-transcript-input` + `codex/hook-processor` 单测 16 passed（集成测试因本地 sqlite3 二进制架构不匹配未跑，属环境问题）。

以下均为不阻断合入的改进项：

- **[Medium] 白名单双份来源**（见行内评论 `codex-transcript-input.ts:25`）：`WAKEUP_RESOURCE_ATTRIBUTE_KEYS` 与共享模块 `DEFAULT_RESOURCE_ENV_FIELD_MAP` 是两份互不感知的真理来源，未来给 env map 增字段会导致 Codex 静默丢弃、与 Qoder 归属维度分叉。建议加注释关联或从共享定义派生。
- **[Low] marker miss 无 debug 日志**（`codex-transcript-input.ts:246-266`）：读取全程 fail-open。若 Codex `session_id` 与 rollout 文件名 UUID 分叉，AgentTeams 归属会静默消失且难排查。建议在 marker 存在但缺 `resourceAttributes` 时打 debug 日志（避免非 AgentTeams 部署刷屏）。
- **[Low] 读取侧缺少 value 长度上限**（`codex-transcript-input.ts:266-270`）：共享采集侧有 512 上限，读取侧无。当前 marker 由自身 hook 写入（已截断），不可利用；建议补对称长度上限。

### Highlights（正向实践）

- Qoder 所有记录路径统一走 `finalizeRecords()`，杜绝「漏打属性」缺陷。
- 三层纵深防敏感信息泄漏，并有针对性的「不泄漏」断言测试。
- wakeup marker 采用临时文件 + 原子 rename，读取与采集全链路 fail-open，不影响宿主 agent 与主采集链路。

---
*评审报告详见: `code-review/pr-114/final-report.md`*
*Generated by LoongSuite-Pilot Code Review Agent*

**Comment by linrunqi08** [comment]
## 🔁 Code Review Round 2 — 复审确认

针对 `41dfd1b9` + `caffbdd6` 两个 fix commit 做了增量复审。上轮 3 个 findings **全部修复**：

| # | 严重度 | 问题 | 状态 |
|---|--------|------|------|
| 1 | Medium | 白名单 `WAKEUP_RESOURCE_ATTRIBUTE_KEYS` 与共享 `DEFAULT_RESOURCE_ENV_FIELD_MAP` 双份来源 | ✅ fixed |
| 2 | Low | marker miss 无 debug 日志 | ✅ fixed |
| 3 | Low | 读取侧缺 value 长度上限 | ✅ fixed |

亮点：Finding #1 没有停留在「加注释」,而是新增了强制对齐测试(`keeps Codex wakeup resource fields aligned with the shared hook env map`),直接 import 共享 map 断言字段透传 —— 未来给共享 map 加字段却漏同步白名单时测试会直接失败,比注释更可靠。Finding #2 也正确地只在「marker 存在但缺属性」时打日志,避免非 AgentTeams 部署刷屏。

**验证**:`npm run typecheck` 通过；`codex-transcript-input`(10→13) + `codex/hook-processor` 共 **19 tests passed**。

**结论:无新增问题、无回归,可以合入。** ✅

---
*Generated by LoongSuite-Pilot Code Review Agent*

File: src/inputs/codex-transcript/codex-transcript-input.ts (hunk 1 above)
Comment (linrunqi08) [reply src/inputs/codex-transcript/codex-transcript-input.ts:27]: ✅ fixed by `41dfd1b9` + `caffbdd6`：已加关联注释,并新增强制对齐测试 `keeps Codex wakeup resource fields aligned with the shared hook env map`(import `DEFAULT_RESOURCE_ENV_FIELD_MAP` 断言全部字段透传)。未来漏同步白名单会直接测试失败。

**Review by linrunqi08 — APPROVED** [review]
